build-requirements.txt

```
@@ -5,6 +5,7 @@ setuptools
 cmake
 ninja
 packaging
+nanobind
```

```suggestion
nanobind>=2.4, <3.0
```
Upstream is more specific about the versions and MLIR at HEAD needs nanobind>=2.4. Let's model this here as well.